File: ruby/lib/sorpresa_ircarcel.rb
module Civitas
    class SorpresaIrCarcel < Sorpresa
        public_class_method :new
        def initialize(tablero)
            super("A la Carcel")
            @tablero=tablero
        end
        
        def aplicarAJugador(actual, todos)
            if jugadorCorrecto(actual, todos)
                informe(actual, todos)
                todos[actual].encarcelar(@tablero.getCarcel)
            end
        end
    end
end

File: Civitas_gui/src/civitas/SorpresaIrCarcel.java
package civitas;

import java.util.ArrayList;

public class SorpresaIrCarcel extends Sorpresa {
    private Tablero tablero;
    SorpresaIrCarcel(Tablero tablero) {
        super("Carcel");
        this.tablero=tablero;
    }

    public void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)){
            informe(actual,todos);
            todos.get(actual).encarcelar(tablero.getCarcel());
        }
    }
}